numpy/lib/tests/test_stride_tricks.py

```diff
@@ -415,12 +415,27 @@ def test_writeable():
     assert_equal(result.flags.writeable, False)
     assert_raises(ValueError, result.__setitem__, slice(None), 0)
 
-    # but the result of broadcast_arrays needs to be writeable (for now), to
+    # but the result of broadcast_arrays needs to be writeable, to
     # preserve backwards compatibility
     for results in [broadcast_arrays(original),
                     broadcast_arrays(0, original)]:
         for result in results:
-            assert_equal(result.flags.writeable, True)
+            # This will change to False in a future version
+            if any([s == 0 for s in result.strides]):
```

Comment:
Simpler as `if 0 in result.strides`?

Reply:
Also, should there be an `else` on this test? What behavior do we expect if no dimensions were broadcast?

I'd be inclined to keep the result warn-on-writeable for consistency.

Reply:
reworked the test. Since `ret = broadcast_arrays([a])` will return `ret[0] is a`, we cannot always set the `warn_on_writeable` flag, that would mean changing the original array.

Reply:
Does `boradcast_arrays(a, a)` return `a` twice by identity?

Reply:
Yes
```
>>> res = np.broadcast_arrays(a, a)
>>> res[0] is a
True
>>> res[1] is a
True
```